transformer_engine/pytorch/jit.py

```diff
@@ -55,18 +63,39 @@ def bgrad_dgelu_fused_(
     return bgrad, dgelu
 
 
+@torch.jit.script
+def dgelu_fused_(
+    grad_output: torch.Tensor, inp: torch.Tensor
+) -> torch.Tensor:
+    """
+    Dgelu fused, this is copy of bgrad_dgelu_fused_ cause jit fusion doesn't allow conditioning.
+    """
+    x = inp
+    tanh_out = torch.tanh(0.79788456 * x * (1 + 0.044715 * x * x))
+    # sqrt(2/pi) * 3 * 0.044715 -> 0.1070322243
+    ff = 0.5 * x * (
+        (1 - tanh_out * tanh_out) * (0.79788456 + 0.1070322243 * x * x)
+    ) + 0.5 * (1 + tanh_out)
+    dgelu = ff * grad_output
+    return dgelu
+
+
 def bias_gelu_fused(inp: torch.Tensor, bias: torch.Tensor) -> torch.Tensor:
     """Disable native AMP for bias_gelu_fused_"""
     with torch.cuda.amp.autocast(enabled=False):
-        return bias_gelu_fused_(inp, bias)
+        if bias.numel() != 0:
+            return bias_gelu_fused_(inp, bias)
+        return gelu_fused_(inp)
 
 
 def bgrad_dgelu_fused(
     grad_output: torch.Tensor, inp: torch.Tensor, bias: torch.Tensor
 ) -> Tuple[torch.Tensor, torch.Tensor]:
     """Disable native AMP for `bgrad_dgelu_fused_`"""
     with torch.cuda.amp.autocast(enabled=False):
-        return bgrad_dgelu_fused_(grad_output, inp, bias)
+        if bias.numel() != 0:
+            return bgrad_dgelu_fused_(grad_output, inp, bias)
+        return torch.zeros_like(bias), dgelu_fused_(grad_output, inp)
```

Comment:
Is this additional allocation necessary? Why don't we return `None` here and change the return type to `Tuple[Union[torch.Tensor, None], torch.Tensor]`?

Reply:
It seems logical to me that `bias` and `bgrad` should have the same type, although I'm not too committed to it. I think the "right" answer would be for `bias` to be `Optional[torch.Tensor]`, but that's probably beyond the scope of this PR.

Reply:
Re: `bias` and `bgrad` should have the same type
I agree with this when a gradient is being computed but that's not the case here right.


Re: the "right" answer would be for `bias` to be `Optional[torch.Tensor]`
While I agree with this it would require modifications at the call site to reflect this.